### ISSUE TITLE
Filter button in data grid must have role, aria-expanded and aria-role for Accessibility

### ISSUE THREAD

**Issue by bhanupradeep7** [vmware/clarity#4110]
## Describe the bug

1)When we navigate to filter button in data grid with the help of keyboard, The screen reader for google chrome reads it as "Grid button column header". 

2)And when clicking on tab to navigate, "filter button" is getting focus before "column header" .

3)The popup input filed that opens up when we click on filter button should have aria-label or aria-labelledby

4) The filter button in the data-grid header does not have aria-label

## How to reproduce

1.  open data-grid page with custom filtering enabled
2.  Click on tab button and navigate to filter button in the column.
3.  Listen to screen reader voice.

## Expected behavior

1) It should read something similar to "Location filter button"-> if we are trying to filter "location" column.
2) Column header should get focus before "filter button".
3) Input field inside popup should have aria-label.
4) Filter button inside data-grid header should have aria-label

## Versions

**App**

* Angular: 8
* Node: 10
* Clarity: 2

**Device:**

* Type: mac
* OS: MAC OS Mojave
* Browser chrome
* Version 78

**Comment by hippee-lee** [comment]
@bhanupradeep7 - am I reading this report correctly to understand that this is using an iPhone + Chrome with a datagrid? Also, the two supported browser+SR combos are

> macOS: Safari + VoiceOVer
> win10: Chrome + Firefox + NVDA

I know there are others, many other combinations. When it comes to triaging and addressing issues we prioritize those two first. I'm not sure if I recall how we discussed handling mobile devices.

@clane - I have a couple of questions about this reported issue. 

1. Should the filter button have specific text like requested? 
2. What is the correct order for getting focus? column text or column filter button?
3. Have we discussed mobile devices + SR combinations?

**Comment by bhanupradeep7** [comment]
I'm Sorry, That was a mistake, It's MAC os Mojave.

**Comment by bashibozuk** [comment]
Are you guys planning to invest efforts on this issue any time soon ?

**Comment by hippee-lee** [comment]
Hi @bashibozuk - this issue is in the backlog but has not been prioritized for someone on the team to pick up at this time.

**Comment by chlane** [comment]
First to address the title "Filter button in data grid must have role, aria-expanded and aria-role for Accessibility" 

The role is for the filter button is addressed in https://github.com/vmware/clarity/issues/3349

aria-expanded  should not be used here because the following are not siblings.

<clr-dg-string-filter _ngcontent-clarity-c230=“”><clr-dg-filter><button type=“button” clrpopoveranchor=“” clrpopoveropenclosebutton=“” class=“datagrid-filter-toggle clr-anchor clr-smart-open-close” aria-label=“Toggle column filter” aria-expanded=“false” aria-controls=“clr-id-9">

<div clrfocustrap="" class="datagrid-filter clr-popover-content ng-star-inserted" tabindex="0" style="top: 189.9px; left: 755px; opacity: 1;" id="clr-id-9"><div class="datagrid-filter-close-wrapper"><button type="button" clrpopoverclosebutton="" class="close clr-smart-close-button">

**Comment by chlane** [comment]
Regarding, "It should read something similar to "Location filter button"-> if we are trying to filter "location" column.".

This is addressed in https://github.com/vmware/clarity/issues/3349

**Comment by chlane** [comment]
Regarding "Column header should get focus before "filter button". " , this is addressed in https://github.com/vmware/clarity/issues/4663

**Comment by chlane** [comment]
Regarding "Input field inside popup should have aria-label.", this is addressed https://github.com/vmware/clarity/issues/4664

**Comment by chlane** [comment]
Regarding, "input field inside popup should have aria-label.".

This is addressed in https://github.com/vmware/clarity/issues/3349

**Comment by gnomeontherun** [comment]
Clarity cares deeply about accessibility, and we work with our accessibility team to identify and triage issues. We have an internal tracking system to identify and verify accessibility issues, which has captured this already. Some issues with Clarity Angular are difficult to solve without causing side effects or breaking changes, and we have to balance concerns with fixing issues and breaking changes. As we build out Clarity Core, we are able to address these root cause accessibility issues and recommend adopting the Clarity Core implementations as they become available. We are closing this issue as we have the issues tracked internally. Please follow our development and releases to see when we release relevant Clarity Core components to make this possible.
